nuitka/freezer/DllDependenciesMacOS.py

```diff
@@ -161,6 +163,14 @@ def _resolveBinaryPathDLLsMacOS(
         elif os.path.basename(path) == os.path.basename(binary_filename):
             # We ignore the references to itself coming from the library id.
             continue
+        elif not os.path.isabs(path) and not os.path.exists(path):
```

Comment:
So, relative paths with any '@' are a thing, where did you see these, how does macOS handle these?!

Reply:
When setting up the linking of one of the necessary dll dependencies for one of the packages that i need for Nuitka-Python, that is the default way that it links it.

Here is the otool Load command:
```
Load command 27
          cmd LC_LOAD_DYLIB
      cmdsize 40
         name libCg.dylib (offset 24)
   time stamp 2 Wed Dec 31 19:00:02 1969
      current version 0.0.0
compatibility version 0.0.0
```

Here is where it searches:
```
dyld[39260]: Library not loaded: 'libCg.dylib'
  Referenced from: '/Users/maxwell/test/python3'
  Reason: tried: 'libCg.dylib' (no such file), '/usr/local/lib/libCg.dylib' (no such file), '/usr/lib/libCg.dylib' (no such file), '/Users/maxwell/test/libCg.dylib' (no such file), '/usr/local/lib/libCg.dylib' (no such file), '/usr/lib/libCg.dylib' (no such file)
```

Reply:
It seems that this happens when a dll is linked in using the `-l` option and the library directory is specified with `-L`. This is really only an issue with Nuitka Python so I updated the code to be in a `isNuitkaPython` check and also to be more precise and robust.